server/src/controllers/leaderboardController.js

```diff
@@ -0,0 +1,15 @@
+import Leaderboard from '../database/models/leaderboard';
+
+const getTopLeaders = async (req, res) => {
+  const topFive = await Leaderboard.find().sort({ score: -1 });
+  const topFiveWithUsers = await Promise.all(topFive.map(async (leaderboard) => {
```

Comment:
@niinpatel @vkbinfo This is the only way I could find to use an async function with `Array.map`. It is very ugly. Could you help me out?

Reply:
Also, I am storing `userId` and `score` in leaderboards. Should I directly store `username` along with `score`?

Reply:
Why do you need user info? You have username and score in the Leaderboard table. Just show the username and score on the leaderboard from LeaderBoard collection.

Reply:
That is what I asked. Right now, there is userId and score in the collection. I will change it to username and score.